### PR TITLE
render task list items with a checkbox

### PR DESCRIPTION
**Pull request by QuietMisdreavus**
Bug/issue #, if applicable: rdar://100145156

## Summary

This PR uses new data from Swift-DocC to render GitHub-style task lists, i.e. list items with a check box.

## Dependencies

https://github.com/apple/swift-docc/pull/388

## Testing

See testing instructions on the linked Swift-DocC PR.

## Checklist

Make sure you check off the following items. If they cannot be completed, provide a reason.

- [x] Added tests
- [x] Ran `npm test`, and it succeeded
- [ n/a ] Updated documentation if necessary

### CODE REVIEW

**Comment by QuietMisdreavus** [comment]
@swift-ci Please test

File: src/components/ContentNode.vue [hunk 1]
```diff
@@ -107,11 +107,29 @@ function renderNode(createElement, references) {
     renderNode(createElement, references),
   );
 
-  const renderListItems = items => items.map(item => (
-    createElement('li', {}, (
-      renderChildren(item.content)
-    ))
-  ));
+  const renderListItems = items => items.map((item) => {
```
Comment (mportiz08) [inline src/components/ContentNode.vue:110]: This change works great, although it adds a bit of complexity to an already very large render function.

Not a blocker, but would you mind if I possibly take a pass at refactoring this to extract an explicit task-list component that we could also re-use for other things in the future if needed?

File: src/styles/base/_typography.scss [hunk 2]
```diff
@@ -108,6 +108,12 @@ li li {
   font-size: 1em;
 }
 
+// make sure that task list checkboxes don't push the rest of the content to the next line
+li input[type="checkbox"] + p {
```
Comment (mportiz08) [inline src/styles/base/_typography.scss:112]: I think we should also remove the default bullet that comes from unordered list items in addition to adding the checkbox. Any objections?

File: src/components/ContentNode.vue (hunk 1 above)
Comment (mportiz08) [reply src/components/ContentNode.vue:110]: I worked on this a bit if you're curious (99.9% done) and can open a PR against your fork if you'd like.

[diff against your branch](https://github.com/QuietMisdreavus/swift-docc-render/compare/task-lists...mportiz08:swift-docc-render:task-list-component)
[diff against main](https://github.com/apple/swift-docc-render/compare/main...mportiz08:swift-docc-render:task-list-component)

File: src/styles/base/_typography.scss (hunk 2 above)
Comment (QuietMisdreavus) [reply src/styles/base/_typography.scss:112]: That's fine by me, though i'm unfamiliar with the CSS needed to accomplish that.

File: src/components/ContentNode.vue (hunk 1 above)
Comment (QuietMisdreavus) [reply src/components/ContentNode.vue:110]: My one reservation about the component as you've written it (if i understand it correctly) is that it's possible to mix "task list items" with plain "list items", e.g. the task list in this PR description. What happens in your patch if someone writes a "mixed list" like that?

Comment (mportiz08) [reply src/components/ContentNode.vue:110]: Ah, I didn't realize it was supposed to be possible to mix/match like that, but I think I realize now why there wasn't a separate list type in the JSON to represent this. I can make one small tweak to not show the checkbox for regular items in that scenario.

I think we would generally want to match the behavior of the GitHub rendering where that item just shows up without the checkbox in that scenario, right?

Example:
- [ ] a
- [x] b
- c
- [ ] d

Comment (mportiz08) [reply src/components/ContentNode.vue:110]: I updated it to work that way with https://github.com/mportiz08/swift-docc-render/commit/c96a2419d6ef1664e990af59a21844c36a21a43d

Comment (QuietMisdreavus) [reply src/components/ContentNode.vue:110]: I think that works, and likely matches what people expect. You can push your branch directly into mine if you'd like.

**Comment by mportiz08** [comment]
@swift-ci test

**Review by mportiz08 — APPROVED** [review]

**Pull request merged**